mono/profiler/proflog.c

```diff
@@ -2490,6 +2541,517 @@ counters_and_perfcounters_sample (MonoProfiler *prof)
 	perfcounters_sample (prof, (now - start) / 1000/ 1000);
 }
 
+#define COVERAGE_DEBUG(x) if (debug_coverage) {x}
+static GHashTable *coverage_methods = NULL;
```

Comment:
It should be easy enough to implement, you just need to take the writer lock while iterating.